### PR TITLE
Fix database variations for ROM

### PR DESCRIPTION
**Pull request by joelvh**
Addresses failing tests in #303 and also removes tests for internals noted in #301

### CODE REVIEW

File: ruby_event_store_rom_sql/spec/rom/event_repository_spec.rb [hunk 1]
```diff
@@ -237,15 +237,15 @@ def cleanup_concurrency_test
     end
 
     def verify_conncurency_assumptions
-      expect(rom_db.connection.pool.size).to eq(5)
+      expect(rom_db.connection.pool.size).to eq(1)
```
Comment (paneq) [inline ruby_event_store_rom_sql/spec/rom/event_repository_spec.rb]: that does not look right. if there is 1 connection in a pool then the multi-threading tests don't make much sense. Is there a way for us to configure more in tests? Like 5 maybe?

Comment (joelvh) [reply ruby_event_store_rom_sql/spec/rom/event_repository_spec.rb]: I wasn’t able to find any Sequel/ROM equivalents to this and not sure of the total context of this test either.

Comment (paneq) [reply ruby_event_store_rom_sql/spec/rom/event_repository_spec.rb]: ah yes, there is:


:preconnect | Whether to automatically make the maximum number of connections when setting up the pool. Can be set to “concurrently” to connect in parallel.

we should use it :)

Comment (joelvh) [reply ruby_event_store_rom_sql/spec/rom/event_repository_spec.rb]: I had looked at the docs, but wasn't sure what specifically was being tested. I haven't played with these settings, but will look into it a bit more.

**Comment by joelvh** [comment]
@paneq added concurrent connections!

**Pull request merged**